src/Resource/Resource.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\WebDav\Resource;
+
+/**
+ * @author Marko Vujnovic <mv@becklyn.com>
+ *
+ * @since  2021-08-13
+ */
+class Resource
+{
+    protected function __construct(protected string $path, protected \DateTimeImmutable $lastModified) {}
```

Comment:
In regards to the conversation above:

```suggestion
    protected function __construct (
        protected string $path,
        protected \DateTimeImmutable $lastModified
    ) {}
```